bin/elasticurl/main.cpp

```diff
@@ -0,0 +1,540 @@
+            case 'o':
+                ctx->output = fopen(aws_cli_optarg, "wb");
+
+                if (!ctx->output)
+                {
+                    fprintf(stderr, "unable to open file %s.\n", aws_cli_optarg);
+                    s_usage(1);
+                }
+                break;
+            case 't':
+                ctx->trace_file = aws_cli_optarg;
+                break;
+            case 'v':
+                if (!strcmp(aws_cli_optarg, "TRACE"))
+                {
+                    ctx->log_level = Aws::Crt::LogLevel::Trace;
+                }
+                else if (!strcmp(aws_cli_optarg, "INFO"))
+                {
+                    ctx->log_level = Aws::Crt::LogLevel::Info;
+                }
+                else if (!strcmp(aws_cli_optarg, "DEBUG"))
+                {
+                    ctx->log_level = Aws::Crt::LogLevel::Debug;
+                }
+                else if (!strcmp(aws_cli_optarg, "ERROR"))
+                {
+                    ctx->log_level = Aws::Crt::LogLevel::Error;
+                }
+                else
+                {
+                    fprintf(stderr, "unsupported log level %s.\n", aws_cli_optarg);
+                    s_usage(1);
+                }
+                break;
+            case 'V':
+                fprintf(stderr, "elasticurl %s\n", ELASTICURL_VERSION);
+                exit(0);
+            case 'w':
+                ctx->alpn = "h2";
+                ctx->required_http_version = AWS_HTTP_VERSION_2;
+                break;
+            case 'W':
+                ctx->alpn = "http/1.1";
+                ctx->required_http_version = AWS_HTTP_VERSION_1_1;
+                break;
+            case 'h':
+                s_usage(0);
+                break;
+            default:
+                fprintf(stderr, "Unknown option\n");
+                s_usage(1);
+        }
+    }
+
+    if (input_body == nullptr)
+    {
+        input_body = std::make_shared<std::stringstream>("");
+    }
+
+    if (aws_cli_optind < argc)
+    {
+        struct aws_byte_cursor uri_cursor = aws_byte_cursor_from_c_str(argv[aws_cli_optind++]);
+
+        if (aws_uri_init_parse(&ctx->uri, ctx->allocator, &uri_cursor))
+        {
+            fprintf(
+                stderr,
+                "Failed to parse uri %s with error %s\n",
+                (char *)uri_cursor.ptr,
+                aws_error_debug_str(aws_last_error()));
+            s_usage(1);
+        };
+    }
+    else
+    {
+        fprintf(stderr, "A URI for the request must be supplied.\n");
+        s_usage(1);
+    }
+}
+
+int main(int argc, char **argv)
+{
+    struct aws_allocator *allocator = aws_default_allocator();
+    (void)allocator;
+    (void)argc;
+    (void)argv;
+
+    struct elasticurl_ctx app_ctx;
+    AWS_ZERO_STRUCT(app_ctx);
+    app_ctx.allocator = allocator;
+    app_ctx.connect_timeout = 3000;
+    app_ctx.output = stdout;
+    app_ctx.verb = "GET";
+    app_ctx.alpn = "h2;http/1.1";
```

Comment:
Just set the initial values for members in the struct definition. Don't zero it out here